nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,450 @@
+""" Read / write access to TCK streamlines format.
+
+TCK format is defined at
+http://mrtrix.readthedocs.io/en/latest/getting_started/image_data.html?highlight=format#tracks-file-format-tck
+"""
+from __future__ import division
+
+import os
+import warnings
+from collections import OrderedDict
+
+import numpy as np
+
+from nibabel.openers import Opener
+from nibabel.py3k import asbytes, asstr
+
+from .array_sequence import ArraySequence
+from .tractogram_file import TractogramFile
+from .tractogram_file import HeaderError, DataWarning, DataError
+from .tractogram import TractogramItem, Tractogram, LazyTractogram
+from .header import Field
+
+MEGABYTE = 1024 * 1024
+BUFFER_SIZE = 1000000
+
+
+def create_empty_header():
+    ''' Return an empty compliant TCK header. '''
+    header = OrderedDict()
+
+    # Default values
+    header[Field.MAGIC_NUMBER] = TckFile.MAGIC_NUMBER
+    header[Field.NB_STREAMLINES] = 0
+    header['datatype'] = "Float32LE"
+    return header
+
+
+class TckFile(TractogramFile):
+    """ Convenience class to encapsulate TCK file format.
+
+    Notes
+    -----
+    MRtrix (so its file format: TCK) considers the streamline coordinate
+    (0,0,0) to be in the center of the voxel which is also the case for
+    NiBabel's streamlines internal representation.
+
+    Moreover, streamlines coordinates coming from a TCK file are considered
+    to be in world space (RAS+ and mm space). MRtrix refers to that space
+    as the "real" or "scanner" space _[1].
+
+    References
+    ----------
+    [1] http://www.nitrc.org/pipermail/mrtrix-discussion/2014-January/000859.html
+    """
+
+    # Contants
+    MAGIC_NUMBER = "mrtrix tracks"
+    SUPPORTS_DATA_PER_POINT = False  # Not yet
+    SUPPORTS_DATA_PER_STREAMLINE = False  # Not yet
+
+    FIBER_DELIMITER = np.array([[np.nan, np.nan, np.nan]], '<f4')
+    EOF_DELIMITER = np.array([[np.inf, np.inf, np.inf]], '<f4')
+
+    def __init__(self, tractogram, header=None):
+        """
+        Parameters
+        ----------
+        tractogram : :class:`Tractogram` object
+            Tractogram that will be contained in this :class:`TckFile`.
+        header : dict (optional)
```

Comment:
No, you are right, it doesn't have to. But maybe the header should be rethinked at some point.